File: 38-CountAndSay/38-CountAndSay.java
// Last updated: 2/13/2026, 10:11:43 AM
class Solution {
    private String countRes(String s){
        StringBuilder res = new StringBuilder();
        int i=0;
        while(i < s.length()){
            int count = 1;
            while (i+1 < s.length() && s.charAt(i) == s.charAt(i+1)){
                count++;
                i++;
            }
            res.append(count).append(s.charAt(i));
            i++;
        }
        return res.toString();
    }    
    public String countAndSay(int n) {
        if(n == 1){
            return "1";
        }
        String res = "1";
        for(int i=0; i<n-1; i++){
            res = countRes(res);
        }
        return res;
    }
}

File: 38-CountAndSay/38-CountAndSay.py
# Last updated: 2/13/2026, 10:11:41 AM
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        countArray = "1"
        def countArr(inp):
            i = 0
            res = ""
            while i < len(inp):
                count = 1
                while i+1 < len(inp) and inp[i] == inp[i+1]:
                    count+=1
                    i+=1
                res+=str(count)+inp[i]
                i+=1
            return res

        for i in range(n-1):
            countArray = countArr(countArray)
        return countArray